overlord/snapstate/backend/aliases.go

```diff
@@ -0,0 +1,123 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package backend
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+)
+
+// Alias represents a command alias with a name and its application target.
+type Alias struct {
+	Name   string `json:"name"`
+	Target string `json:"target"`
+}
+
+// MissingAliases returns the subset of aliases that are missing on disk.
+func (b Backend) MissingAliases(aliases []*Alias) ([]*Alias, error) {
+	var res []*Alias
+	for _, cand := range aliases {
+		_, err := os.Lstat(filepath.Join(dirs.SnapBinariesDir, cand.Name))
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return nil, err
+			}
+			res = append(res, cand)
```

Comment:
Shouldn't we check if it matches here?  Alternatively, we should probably rename the function to `MissingNames` or similar and fix the docs, since the actual alias being passed in can be missing.